tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1637 @@
+
+        if not self._validate_media_type(req, resp):
+            self.wlogger.warning("Media type invalid in produce request")
+            return
+        elif self._check_active_thread(self.produce_thread, req, resp):
+            self.wlogger.warning("Produce action is active")
+            return
+
+        # update topic config
+        topics_cfg_keys = list(vars(self.workload_config).keys())
+        topics_cfg_keys += ["msg_rate_limit", "msg_total"]
+        if not self._validate_request(topics_cfg_keys, [], req, resp):
+            # response is already populated inside validate_request
+            return
+        else:
+            # Update topic configs
+            self.message_rate = req.media['msg_rate_limit'] \
+                if 'msg_rate_limit' in req.media else app_config.msg_rate_limit
+            self.total_messages = req.media['msg_total'] \
+                if 'msg_total' in req.media else app_config.msg_total
+            if 'topic_prefix_produce' in req.media:
+                self.workload_config.topic_prefix_produce = req.media[
+                    'topic_prefix_produce']
+            if 'topic_count' in req.media:
+                self.workload_config.topic_count = req.media['topic_count']
+            # start producers
+            self.init_producers()
+            self.produce(wait=False)
+
+            resp.status = falcon.HTTP_200
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {'result': "OK"}
+
+
+class StreamVerifierConsume(StreamVerifierWeb):
+    """Handled host:port/consume requests
+
+    On POST runs consume command, can use cfg from status JSON.
+    {
+        "topic_group_id": "group-stream-verifier-tx",
+        "topic_prefix_produce": "stream-topic-src",
+        "topic_prefix_consume": "stream-topic-dst",
+        "topic_count": 16,
+        "consume_timeout_s": 60,
+        "msg_rate_limit": 0,
+        "msg_total": 256
+    }
+
+    If Consume is active, returns error JSON
+    Validation errors are returned in the same way
+    {
+        "errors": [
+            "Active consume job not finished"
+        ]
+    }
+
+    """
+    def on_post(self, req: falcon.Request, resp: falcon.Response):
+        self.wlogger.debug("Processing consume post request")
+        # Validate request
+        if not self._validate_media_type(req, resp):
+            self.wlogger.warning("Media type invalid in produce request")
+            return
+        elif self._check_active_thread(self.consume_thread, req, resp):
+            self.wlogger.warning("Consume action is active")
+            return
+
+        # update topic config
+        topics_cfg_keys = ["topic_prefix_consume", "topic_count"]
+        forbidden = ["msg_rate_limit", "msg_total"]
+        if not self._validate_request(topics_cfg_keys, forbidden, req, resp):
+            # response is already populated inside validate_request
+            return
+        else:
```

Comment:
No need for the `else:` here, really, is there? Could unindent the rest of the block below.

Reply:
Well, it is a habit after C lang. Code written like that eliminates couple of compiler generated jumps.
I'll update that since it'l look better, I guess.